### PR TITLE
fix(apigateway): api key not supported for SpecRestApi

### LINKED ISSUE #11079: [aws-apigateway] ApiKey is not supported for SpecRestApi
There's an API disparity for `ApiKey` between `RestApi` and `SpecRestApi` constructs:

https://github.com/aws/aws-cdk/blob/2b474b9ba6a2c2ff4e79e565ba272bb4029190d0/packages/%40aws-cdk/aws-apigateway/lib/api-key.ts#L48-L54

Notice how `resources` is of type `RestApi[]` instead of `IRestApi[]`, making this impossible:

```ts
class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    const api = new SpecRestApi(this, "MyApi", {
      apiDefinition: ApiDefinition.fromInline({ /* (... )*/ }),
    });
    new ApiKey(this, "MyApi", {
      // ERROR: Type 'SpecRestApi' is missing the following properties from type 'RestApi': methods, deployments, url, addApiKey, and 2 more.
      resources: [api],
    });
  }
}
```

Additionally, the convenience method:

https://github.com/aws/aws-cdk/blob/2b474b9ba6a2c2ff4e79e565ba272bb4029190d0/packages/%40aws-cdk/aws-apigateway/lib/restapi.ts#L712-L717

Is only available in `RestApi`. It should probably be moved to `RestApiBase` so `SpecRestApi` can benefit from it as well.

Fortunately, this disparity doesn't make it impossible to create/associate an API key to an API:

```ts
class MyStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps) {
    super(scope, id, props);

    const api = new SpecRestApi(this, "MyApi", {
      apiDefinition: ApiDefinition.fromInline({}),
    });

    // option 1
    new UsagePlan(this, "MyUsagePlan", {
      apiKey: new ApiKey(this, "MyApi"),
      apiStages: [{ stage: api.deploymentStage }],
    });

    // option 2
    api.addUsagePlan("MyUsagePlan", {
      apiStages: [{ stage: api.deploymentStage }],
    });
  }
}
```

As a side note, `option 2` from the previous example has an unnecessary requirement to assign a value to `apiStages` in order to bind it to the API.

I believe this could be improved by changing:

https://github.com/aws/aws-cdk/blob/2b474b9ba6a2c2ff4e79e565ba272bb4029190d0/packages/%40aws-cdk/aws-apigateway/lib/restapi.ts#L350-L352

to:

```ts
public addUsagePlan(id: string, props: UsagePlanProps = {}): UsagePlan {
  return new UsagePlan(this, id, {
    apiStages: [
      { stage: this.deploymentStage },
      ...(props.apiStages?.filter((x) => x.stage != this.deploymentStage) ||
        []),
    ],
    ...props,
  });
}
```

### Reproduction Steps

_See description above_

### What did you expect to happen?

1. `RestApi` and `SpecRestApi` have API parity for dealing with API keys.
1. Calling `RestApiBase.prototype.addUsagePlan()` automatically binds the usage plan to the `RestApiBase` instance.

### What actually happened?

1. The APIs are different.
1. Calling `RestApiBase.prototype.addUsagePlan()` does not bind the usage plan to the `RestApiBase` instance. It has to be specified in the `apiStages` field of the `UsagePlanProps` parameter.

### Environment

  - **CLI Version:** `1.68.0`
  - **Framework Version:** `1.69.0`
  - **Node.js Version:** `12.13.0`
  - **OS:** macOS `10.14.6`
  - **Language (Version):** `typescript@4.0.3`

### Other

N/A

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by ayush987goyal**
fix(apigateway): ApiKey not supported for SpecRestApi

closes #11079 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by drissamri** [comment]
This would be a great addition so we don't have to go through an usage plan 👍

File: packages/@aws-cdk/aws-apigateway/lib/api-key.ts [hunk 1]
```diff
@@ -51,7 +51,7 @@ export interface ApiKeyProps extends ApiKeyOptions {
    * A list of resources this api key is associated with.
    * @default none
    */
-  readonly resources?: RestApi[];
```
Comment (nija-at) [inline packages/@aws-cdk/aws-apigateway/lib/api-key.ts:54]: You can now remove the line `[disable-awslint:...` in the comment above.

**Review by nija-at — CHANGES_REQUESTED** [review]
Nice! Just one small comment.

File: packages/@aws-cdk/aws-apigateway/lib/api-key.ts (hunk 1 above)
Comment (ayush987goyal) [reply packages/@aws-cdk/aws-apigateway/lib/api-key.ts:54]: Done!

**Review by nija-at — DISMISSED** [review]

**Comment by ayush987goyal** [comment]
@nija-at A re-approval is required since I had to rebase from master for build to succeed.

**Review by nija-at — APPROVED** [review]

**Pull request merged**